app/routes/individual_response.py

```diff
@@ -248,7 +250,12 @@ def individual_response_text_message_confirmation(schema, questionnaire_store):
     if request.method == "POST":
         return redirect(url_for("questionnaire.get_questionnaire"))
 
-    mobile_number = url_safe_serializer().loads(request.args.get("mobile_number"))
+    if not (request_mobile_number := request.args.get("mobile_number")):
+        raise BadRequest
+    try:
+        mobile_number = url_safe_serializer().loads(request_mobile_number)
+    except BadSignature:
+        raise BadRequest
```

Comment:
This works without checking for the existence of the `mobile_number` request argument:
```suggestion
    try:
        mobile_number = url_safe_serializer().loads(request.args["mobile_number"])
    except BadSignature:
        raise BadRequest
```
A `werkzeug.exceptions.BadRequestKeyError` is raised and caught by the 400 error handler. For clarity we could add `BadRequestKeyError` to the 400 error handler.

This also has the advantage that the error trace logging makes a clear distinction between the key error and the bad signature.